File: Auto-Part/FrontEnd/Seller/src/app/services/diliver.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Diliver} from "../dtos/diliver";
import {Observable} from "rxjs";
import {MAIN_URL} from "./url.service";

const URL="/api/v1/diliver";

@Injectable()
export class DiliverService {

  constructor(private http:HttpClient) { }

  save(diliver:Diliver):Observable<boolean>{
    return this.http.post<boolean>(MAIN_URL+URL+"/",diliver);
  }

  delete(dNic:string):Observable<boolean>{
    return this.http.delete<boolean>(MAIN_URL+URL+"/"+dNic);
  }

  getAllDiliver():Observable<Array<Diliver>>{
    return this.http.get<Array<Diliver>>(MAIN_URL+URL);
  }

  getAllDilivers():Observable<number>{
    return this.http.get<number>(MAIN_URL+URL+"/count")
  }
}

File: Auto-Part/BackEnd/src/main/java/lk/ijse/autopart/rest/service/DiliverService.java
package lk.ijse.autopart.rest.service;

import lk.ijse.autopart.rest.dto.DiliverDTO;

import java.util.ArrayList;

public interface DiliverService {

    public boolean save(DiliverDTO diliverDTO);

    public boolean delete(String dNic);

    public ArrayList<DiliverDTO>getAllDiliver();

    public DiliverDTO getDiliver(String dNic);

    public long getAllDilivers();
}
